storage/store.go

```diff
@@ -486,6 +483,19 @@ func (s *Store) Start(stopper *util.Stopper) error {
 		if err := s.startGossip(); err != nil {
 			return err
 		}
+
+		// Start the scanner. The construction here makes sure that the scanner
+		// only starts after Gossip has connected, and that it does not block Start
+		// from returning (as doing so might prevent Gossip from ever connecting).
+		s.stopper.RunWorker(func() {
+			select {
+			case <-s.ctx.Gossip.Connected:
+				s.scanner.Start(s.ctx.Clock, s.stopper)
+			case <-s.stopper.ShouldStop():
+				return
```

Comment:
That would also work, but it seems less straightforward (Gossip could be nil, the channel seems weird from the perspective of the scanner). I think this is a good place as this issue is inherently store-bootstrap-related.